turbinia/workers/analysis/jenkins.py

```diff
@@ -0,0 +1,160 @@
+
+    try:
+      collected_artifacts = extract_artifacts(
+        artifact_names=['JenkinsConfigFile'],
+        disk_path=evidence.local_path,
+        output_dir=os.path.join(self.output_dir, 'artifacts')
+      )
+    except RuntimeError as e:
+      result.close(self, success=False, status=str(e))
+      return result
+
+    version = None
+    credentials = []
+    for filepath in collected_artifacts:
+        with open(filepath, 'r') as input_file:
+          config = input_file.read()
+
+        extracted_version = self._extract_jenkins_version(config)
+        extracted_credentials = self._extract_jenkins_credentials(config)
+
+        if extracted_version:
+          version = extracted_version
+
+        credentials.extend(extracted_credentials)
+
+    analysis_report = self.analyse_jenkins(version, credentials)
+    output_evidence.text_data = analysis_report
+
+    # Write the report to the output file.
+    with open(output_file_path, 'w') as fh:
+      fh.write(output_evidence.text_data.encode('utf8'))
+      fh.write('\n'.encode('utf8'))
+
+    # Add the resulting evidence to the result object.
+    result.add_evidence(output_evidence, evidence.config)
+    result.close(self, success=True)
+
+    return result
+
+  @staticmethod
+  def _extract_jenkins_version(config):
+    """Extract version from Jenkins configuration files.
+
+    Args:
+      config (str): configuration file content.
+
+    Returns:
+      str: The version of Jenkins.
+    """
+    version = None
+    version_re = re.compile('<version>(.*)</version>')
+    version_match = re.search(version_re, config)
+
+    if version_match:
+      version = version_match.group(1)
+
+    return version
+
+  @staticmethod
+  def _extract_jenkins_credentials(config):
+    """Extract credentials from Jenkins configuration files.
+
+    Args:
+      config (str): configuration file content.
+
+    Returns:
+      list: of tuples with username and password hash.
+    """
+    credentials = []
+    password_hash_re = re.compile('<passwordHash>#jbcrypt:(.*)</passwordHash>')
+    username_re = re.compile('<fullName>(.*)</fullName>')
+
+    password_hash_match = re.search(password_hash_re, config)
+    username_match = re.search(username_re, config)
+
+    if username_match and password_hash_match:
+      username = username_match.group(1)
+      password_hash = password_hash_match.group(1)
+      credentials.append((username, password_hash))
+
+    return credentials
+
+  @staticmethod
+  def analyse_jenkins(version, credentials):
+    """Analyses a Jenkins configuration.
+
+    Args:
+      version (str): Version of Jenkins.
+      credentials (list): of tuples with username and password hash.
+
+    Returns:
+      str: of description of security of Jenkins configuration file.
+    """
+
+    findings = []
+    credentials_registry = {hash: username for username, hash in credentials}
+    weak_passwords = bruteforce_password_hashes(credentials_registry.keys())
```

Comment:
Can you add a TODO here to set a timeout method parameter here that we can override with "dynamic task configuration" once it's available? That's part of the "Recipes" design that we talked about.

Reply:
I just filed https://github.com/google/turbinia/issues/244 that you can reference here as well.

Reply:
Done.